db/commitlog/commitlog.cc

```diff
@@ -396,6 +397,21 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
         }
     }
 
+    future<> force_sync(time_point latest) {
+        if (_timer.get_timeout() < latest) {
+            return make_ready_future<>();
+        }        
+        
+        _timer.cancel();
+        
+        if (clock_type::now() < latest) {
+            on_timer();
```

Comment:
No need for this, if the timer is already expired by the time you arm it, it will fire next reactor poll. I think it's better to have fewer paths into the timer callback.

Reply:
Removed